tests/test_modeling_common.py

```diff
@@ -2476,6 +2553,44 @@ def test_checkpoint_sharding_from_hub(self):
         for p1, p2 in zip(model.parameters(), ref_model.parameters()):
             self.assertTrue(torch.allclose(p1, p2))
 
+    def test_from_pretrained_low_cpu_mem_usage_measured(self):
```

Comment:
If you can make it work consistently, I would put this one as slow (I think I saw a comment about that in an eralier version, so that might be your plan ;-) )

Reply:
yes, it will be slow - I just wanted to see how CI runs those so will add at the end.

I'm revamping the functionality and came up with a slightly more reliable idea for tests.

Still need a lot of cleanup, so please ignore for now.